internal/workflow/batch.go

```diff
@@ -32,13 +35,53 @@ func NewBatchWorkflow(
 }
 
 func (w *BatchWorkflow) Execute(ctx temporalsdk_workflow.Context, req *ingest.BatchWorkflowRequest) (e error) {
-	logger := temporalsdk_workflow.GetLogger(ctx)
-	logger.Info(
+	state := newBatchWorkflowState(ctx, req)
+	state.logger.Info(
 		"Starting batch workflow",
 		"batch_uuid", req.Batch.UUID.String(),
 		"batch_identifier", req.Batch.Identifier,
 		"source_id", req.SIPSourceID.String(),
 		"keys", req.Keys,
 	)
+
+	defer func() {
+		// Update final batch status and add completion date.
+		state.batch.CompletedAt = temporalsdk_workflow.Now(ctx)
+		state.batch.Status = enums.BatchStatusIngested
+		if e != nil {
+			state.batch.Status = enums.BatchStatusFailed
+		}
+		if err := w.updateBatch(ctx, state); err != nil {
+			e = errors.Join(e, err)
+		}
+	}()
+
+	// Update batch status to Processing and add start date.
```

Comment:
Nitpick: the actual value of "processing" is not capitalized, so I would quote the actual value here, e.g.

```
// Update batch status to "processing" and add start date.
```